encode.js

```diff
@@ -0,0 +1,115 @@
+const { readdirSync, readFileSync, writeFileSync } = require("fs");
+const { parse } = require("yaml");
+
+// get yaml files in this repo
+const files = readdirSync(__dirname).filter((file) => /\.ya?ml$/.test(file));
+
+// start master list of redirects
+const list = [];
+
+// save errors for reporting later
+const errors = [];
+
+// go through each yaml file
+for (const file of files) {
+  // load file contents
+  const contents = readFileSync(file, "utf8");
+
+  // try to parse as yaml
+  let data;
+  try {
+    data = parse(contents);
+  } catch (error) {
+    errors.push(`Couldn't parse ${file}. Make sure it is valid YAML.`);
+    continue;
+  }
+
+  // check if top level is list
+  if (!Array.isArray(data)) {
+    errors.push(`${file} is not a list`);
+    continue;
+  }
+
+  // go through each entry
+  for (let [index, entry] of Object.entries(data)) {
+    index = Number(index) + 1;
+    const log = `${file} entry ${index}`;
+
+    // check if dict
+    if (typeof entry !== "object") {
+      errors.push(`${log} is not a dict`);
+      continue;
+    }
+
+    // check "from" field
+    if (!(typeof entry.from === "string" && entry.from.trim())) {
+      errors.push(`${log} "from" field invalid`);
+      continue;
+    }
+
+    // check "to" field
+    if (!(typeof entry.to === "string" && entry.to.trim()))
+      errors.push(`${log} "to" field invalid`);
+
+    // normalize "from" field. lower case, remove leading slashes.
+    entry.from = entry.from.toLowerCase().replace(/^(\/+)/, "");
```

Comment:
This is a super minor nitpick, but URLs are case-sensitive, and the end user might be surprised when their case isn't retained. You might want to mention that the "from" field is made lowercase or remove the case insensitivity.

Reply:
This is by design and request. This approach is intended to be as simple and error-proof as possible. I can easily envision people going to `/TimZoom` and being frustrated when nothing happens because they wanted `/timzoom`.

I've added a bolded note to the edit instructions about the case-insensitivity.